test/correctness/external_code.cpp

```diff
@@ -0,0 +1,60 @@
+#include <stdio.h>
+#include "Halide.h"
+#include <fstream>
+#include <cassert>
+#include <iostream>
+
+#include "test/common/halide_test_dirs.h"
+
+using namespace Halide;
+
+int main(int argc, char **argv) {
+    Var x("x"), y("y");
+    Func f("f");
+
+    f(x, y) = 42;
+
+    std::string bitcode_file = Internal::get_test_tmp_dir() + "extern.bc";
+    f.compile_to_bitcode(bitcode_file, {}, "extern");
+
+    std::vector<uint8_t> bitcode;
+    std::ifstream bitcode_stream(bitcode_file, std::ios::in | std::ios::binary);
+    bitcode_stream.seekg(0, std::ios::end);
+    bitcode.resize(bitcode_stream.tellg());
+    bitcode_stream.seekg(0, std::ios::beg);
+    bitcode_stream.read(reinterpret_cast<char *>(&bitcode[0]), bitcode.size());
+
+    ExternalCode external_code = ExternalCode::bitcode_wrapper(get_target_from_environment(), bitcode, "extern");
+
+    Func f_extern;
+    f_extern.define_extern("extern", { }, type_of<int32_t>(), 2);
+
+    Func result;
+    result(x, y) = f_extern(x, y);
+
+    Module module = result.compile_to_module({}, "forty_two");
+
+    module.append(external_code);
+
+    auto forty_two = module.get_function_by_name("forty_two");
+
+    Internal::JITModule jit_module(module, forty_two, {});
+    
+    auto argv_wrapper = jit_module.argv_function();
+
+    Buffer<int32_t> buf(16, 16);
+    const void *args[1];
```

Comment:
Or given that there's one arg, you could also just grab the main function and do a function pointer cast on it to int (*)(halide_buffer_t *)